test/test_did-change-watched-files.jl

```diff
@@ -0,0 +1,177 @@
+module test_didchangewatchedfiles
+
+using Test
+using JETLS
+using TOML
+
+include("setup.jl")
+
+# `workspace/didChangeWatchedFiles` does not support static registration,
+# so we need to allow dynamic registration here to test it
+const CLIENT_CAPABILITIES = ClientCapabilities(
+    workspace = (;
+        didChangeWatchedFiles = DidChangeWatchedFilesClientCapabilities(
+            dynamicRegistration = true
+        )
+    )
+)
+
+@testset "DidChangeWatchedFilesNotification full-cycle" begin
+    mktempdir() do tmpdir
+        config_path = joinpath(tmpdir, ".JETLSConfig.toml")
+        open(config_path, "w") do io
+            write(io, "[performance.full_analysis]\ndebounce = 2.0\n[testrunner]\nexecutable = \"mytestrunner\"\n")
+        end
+        rootUri = filepath2uri(tmpdir)
+        withserver(; rootUri, capabilities=CLIENT_CAPABILITIES) do (; writereadmsg, readmsg, id_counter, server)
+            @test server.state.config_manager.actual_config["performance"]["full_analysis"]["debounce"] == 2.0
+            @test server.state.config_manager.latest_config["performance"]["full_analysis"]["debounce"] == 2.0
+
+            @test server.state.config_manager.actual_config["testrunner"]["executable"] == "mytestrunner"
+            @test server.state.config_manager.latest_config["testrunner"]["executable"] == "mytestrunner"
+
+            open(config_path, "w") do io
+                write(io, "[performance.full_analysis]\ndebounce = 3.0\n[testrunner]\nexecutable = \"mytestrunner\"\n")
```

Comment:
```suggestion
                write(io, """
                [performance.full_analysis]
                debounce = 3.0
                
                [testrunner]
                executable = "mytestrunner"
                """)
```
may be easier to read?